tests/integration/test_load_and_run_multi_agent_a2a_tool.py

```diff
@@ -0,0 +1,206 @@
+import asyncio
+import logging
+import os
+
+import pytest
+from litellm.utils import validate_environment
+from rich.logging import RichHandler
+
+from any_agent import AgentConfig, AgentFramework, AnyAgent
+from any_agent.config import ServingConfig, TracingConfig
+from any_agent.tools import a2a_query
+from any_agent.tracing.agent_trace import AgentSpan, AgentTrace
+
+FORMAT = "%(message)s"
+logging.basicConfig(
+    level=logging.DEBUG,
+    format=FORMAT,
+    datefmt="[%X]",
+    handlers=[RichHandler(rich_tracebacks=True)],
+)
+logger = logging.getLogger("any_agent_test")
+logger.setLevel(logging.DEBUG)
+
+CHILD_TAG = "any_agent.children"
+
+
+def organize(items: list[AgentSpan]) -> None:
+    traces = {}
+    for trace in items:
+        k = trace.context.span_id
+        trace.attributes[CHILD_TAG] = {}
+        traces[k] = trace
+    for trace in items:
+        if trace.parent:
+            parent_k = trace.parent.span_id
+            if parent_k:
+                traces[parent_k].attributes[CHILD_TAG][trace.context.span_id] = trace
+            else:
+                traces[None] = trace
+    logger.info(traces[None].model_dump_json(indent=2))
+
+
+@pytest.mark.skipif(
+    os.environ.get("ANY_AGENT_INTEGRATION_TESTS", "FALSE").upper() != "TRUE",
+    reason="Integration tests require `ANY_AGENT_INTEGRATION_TESTS=TRUE` env var",
+)
+@pytest.mark.asyncio
+async def test_load_and_run_multi_agent(
+    agent_framework: AgentFramework,
+) -> None:
+    """Tests that an agent contacts another using A2A using the adapter tool.
+
+    Note that there is an issue when using Google ADK: https://github.com/google/adk-python/pull/566
+    """
+    kwargs = {}
+
+    kwargs["model_id"] = "gpt-4.1-nano"
+    agent_model = kwargs["model_id"]
+    env_check = validate_environment(kwargs["model_id"])
+    if not env_check["keys_in_environment"]:
+        pytest.skip(f"{env_check['missing_keys']} needed for {agent_framework}")
+
+    model_args = (
+        {"parallel_tool_calls": False}
+        if agent_framework not in [AgentFramework.AGNO, AgentFramework.LLAMA_INDEX]
+        else None
+    )
+
+    main_agent = None
+    served_agent = None
+    served_task = None
+
+    try:
+        tool_agent_port = 5800
+        tool_agent_endpoint = "tool_agent"
+
+        # SEARCH AGENT
+
+        """
```

Comment:
Why is the commented code left here?

Reply:
I'll remove it. I wanted to use a simpler alternative than search_web but I thought I would add the other tool later :-/ I don't think it adds anything to the tests right now.